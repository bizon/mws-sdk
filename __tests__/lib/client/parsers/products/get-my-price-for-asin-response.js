const parseXml = require('../../../../../lib/client/parsers')
const parseGetMyPriceForAsinResponse = require('../../../../../lib/client/parsers/products/get-my-price-for-asin-response')

describe('lib.client.parsers.products.get-my-price-for-asin-response', () => {
  it('should parse the GetMyPriceForASINResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetMyPriceForASINResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetMyPriceForASINResult ASIN="1933890517" status="Success">
          <Product xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01"
            xmlns:ns2="http://mws.amazonservices.com/schema/Products/2011-10-01/default.xsd">
            <Identifiers>
              <MarketplaceASIN>
                <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
                <ASIN>1933890517</ASIN>
              </MarketplaceASIN>
            </Identifiers>
            <Offers>
              <Offer>
                <BuyingPrice>
                  <LandedPrice>
                    <CurrencyCode>USD</CurrencyCode>
                    <Amount>303.99</Amount>
                  </LandedPrice>
                  <ListingPrice>
                    <CurrencyCode>USD</CurrencyCode>
                    <Amount>300.00</Amount>
                  </ListingPrice>
                  <Shipping>
                    <CurrencyCode>USD</CurrencyCode>
                    <Amount>3.99</Amount>
                  </Shipping>
                </BuyingPrice>
                <RegularPrice>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>300.00</Amount>
                </RegularPrice>
                <FulfillmentChannel>MERCHANT</FulfillmentChannel>
                <ItemCondition>Used</ItemCondition>
                <ItemSubCondition>Acceptable</ItemSubCondition>
                <SellerId>A1IMEXAMPLEWRC</SellerId>
                <SellerSKU>SKU2468</SellerSKU>
              </Offer>
            </Offers>
          </Product>
        </GetMyPriceForASINResult>
        <ResponseMetadata>
          <RequestId>a3381684-87bd-416e-9b95-EXAMPLE9c236</RequestId>
        </ResponseMetadata>
      </GetMyPriceForASINResponse>`,
    )

    const res = parseGetMyPriceForAsinResponse('/products:GetMyPriceForASINResponse', doc)

    expect(res).toMatchSnapshot()
  })

  it('should parse GetMyPriceForASINResponse errors', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetMyPriceForASINResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetMyPriceForASINResult ASIN="1933890517" status="Success">
          <Error>
            <Type>Sender</Type>
            <Code>InvalidParameterValue</Code>
            <Message>1933890517 is an invalid ASIN for marketplace ATVPDKIKX0DER</Message>
          </Error>
        </GetMyPriceForASINResult>
        <ResponseMetadata>
          <RequestId>a3381684-87bd-416e-9b95-EXAMPLE9c236</RequestId>
        </ResponseMetadata>
      </GetMyPriceForASINResponse>`,
    )

    const res = parseGetMyPriceForAsinResponse('/products:GetMyPriceForASINResponse', doc)

    expect(res).toMatchSnapshot()
  })
})
