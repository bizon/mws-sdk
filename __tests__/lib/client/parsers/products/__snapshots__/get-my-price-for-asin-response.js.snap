// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.products.get-my-price-for-asin-response should parse GetMyPriceForASINResponse errors 1`] = `
{
  "getMyPriceForAsinResult": [
    {
      "asin": "1933890517",
      "error": {
        "code": "InvalidParameterValue",
        "detail": null,
        "message": "1933890517 is an invalid ASIN for marketplace ATVPDKIKX0DER",
        "type": "Sender",
      },
      "product": null,
      "status": "Success",
    },
  ],
  "responseMetadata": {
    "requestId": "a3381684-87bd-416e-9b95-EXAMPLE9c236",
  },
}
`;

exports[`lib.client.parsers.products.get-my-price-for-asin-response should parse the GetMyPriceForASINResponse example response from MWS doc 1`] = `
{
  "getMyPriceForAsinResult": [
    {
      "asin": "1933890517",
      "error": null,
      "product": {
        "attributes": null,
        "identifiers": {
          "marketplaceAsin": {
            "asin": "1933890517",
            "marketplaceId": "ATVPDKIKX0DER",
          },
          "skuIdentifier": null,
        },
        "offers": [
          {
            "buyingPrice": {
              "landedPrice": {
                "amount": 303.99,
                "currencyCode": "USD",
              },
              "listingPrice": {
                "amount": 300,
                "currencyCode": "USD",
              },
              "shipping": {
                "amount": 3.99,
                "currencyCode": "USD",
              },
            },
            "fulfillmentChannel": "MERCHANT",
            "itemCondition": "Used",
            "itemSubCondition": "Acceptable",
            "regularPrice": {
              "amount": 300,
              "currencyCode": "USD",
            },
            "sellerId": "A1IMEXAMPLEWRC",
            "sellerSku": "SKU2468",
          },
        ],
        "relationships": {
          "variationChildren": [],
          "variationParent": null,
        },
        "salesRankings": [],
      },
      "status": "Success",
    },
  ],
  "responseMetadata": {
    "requestId": "a3381684-87bd-416e-9b95-EXAMPLE9c236",
  },
}
`;
